### DISCUSSION TITLE
Crash after clicking "Report an Issue"

### DISCUSSION THREAD

**Discussion by pravdomil** [keiwando/toonsquid#211] [Ideas]
You should have it in crash longs. I'm using Chrome as default browser on iPad.

> maybe rename rename github discussion topic ideas to feedback

**Comment by keiwando** [comment]
Thanks for reporting this (and for all of your other feedback!)

I've heard of this crash from some users but so far I haven't been able to reproduce it on any of my iPads at all. I am unfortunately also not seeing any related crash log. Maybe you can help me narrow down the cause of it. Could you please check if you can locate the crash log locally on your iPad in the settings app (Analytics & Improvements > Analytics Data) and send that to contact@toonsquid.com?

Does it always crash when you tap on "Report an issue" or does it work sometimes? It's supposed to open a partially pre-filled native e-mail dialog.

**Reply by pravdomil** [reply]
it crashes always :D

[for me it looks like calling `presentViewController` from non main thread](https://stackoverflow.com/questions/58084441/ios-crash-in-uiviewcontroller-presentviewcontroller)

let me know if you need more of the crash log 

<details>
<summary>crash log</summary>

```json
{
  "faultingThread": 0,
  "exception": {
    "codes": "0x0000000000000000, 0x0000000000000000",
    "rawCodes": [
      0,
      0
    ],
    "type": "EXC_CRASH",
    "signal": "SIGABRT"
  },
  "asi": {
    "libsystem_c.dylib": [
      "abort() called"
    ]
  },
  "lastExceptionBacktrace": [
    {
      "imageOffset": 39460,
      "symbol": "__exceptionPreprocess",
      "symbolLocation": 160,
      "imageIndex": 6
    },
    {
      "imageOffset": 88408,
      "symbol": "objc_exception_throw",
      "symbolLocation": 56,
      "imageIndex": 4
    },
    {
      "imageOffset": 3151120,
      "symbol": "-[UIViewController _presentViewController:withAnimationController:completion:]",
      "symbolLocation": 4272,
      "imageIndex": 8
    },
    {
      "imageOffset": 3146684,
      "symbol": "__63-[UIViewController _presentViewController:animated:completion:]_block_invoke",
      "symbolLocation": 88,
      "imageIndex": 8
    },
    {
      "imageOffset": 3012388,
      "symbol": "-[UIViewController _performCoordinatedPresentOrDismiss:animated:]",
      "symbolLocation": 364,
      "imageIndex": 8
    },
    {
      "imageOffset": 824276,
      "symbol": "-[UIViewController _presentViewController:animated:completion:]",
      "symbolLocation": 160,
      "imageIndex": 8
    },
    {
      "imageOffset": 823884,
      "symbol": "-[UIViewController presentViewController:animated:completion:]",
      "symbolLocation": 140,
      "imageIndex": 8
    },
    {
      "imageOffset": 4792200,
      "imageIndex": 12
    },
    {
      "imageOffset": 512984,
      "imageIndex": 12
    },
    {
      "imageOffset": 506256,
      "imageIndex": 12
    },
    {
      "imageOffset": 4168024,
      "symbol": "-[UIAlertController _invokeHandlersForAction:]",
      "symbolLocation": 80,
      "imageIndex": 8
    },
    {
      "imageOffset": 5698348,
      "symbol": "__103-[UIAlertController _dismissAnimated:triggeringAction:triggeredByPopoverDimmingView:dismissCompletion:]_block_invoke_2",
      "symbolLocation": 28,
      "imageIndex": 8
    },
    {
      "imageOffset": 2485740,
      "symbol": "-[UIPresentationController transitionDidFinish:]",
      "symbolLocation": 1076,
      "imageIndex": 8
    },
    {
      "imageOffset": 7335996,
      "symbol": "__56-[UIPresentationController runTransitionForCurrentState]_block_invoke.88",
      "symbolLocation": 316,
      "imageIndex": 8
    },
    {
      "imageOffset": 2712564,
      "symbol": "-[_UIViewControllerTransitionContext completeTransition:]",
      "symbolLocation": 104,
      "imageIndex": 8
    },
    {
      "imageOffset": 15998316,
      "symbol": "__UIVIEW_IS_EXECUTING_ANIMATION_COMPLETION_BLOCK__",
      "symbolLocation": 28,
      "imageIndex": 8
    },
    {
      "imageOffset": 819768,
      "symbol": "-[UIViewAnimationBlockDelegate _didEndBlockAnimation:finished:context:]",
      "symbolLocation": 604,
      "imageIndex": 8
    },
    {
      "imageOffset": 815596,
      "symbol": "-[UIViewAnimationState sendDelegateAnimationDidStop:finished:]",
      "symbolLocation": 212,
      "imageIndex": 8
    },
    {
      "imageOffset": 813564,
      "symbol": "-[UIViewAnimationState animationDidStop:finished:]",
      "symbolLocation": 192,
      "imageIndex": 8
    },
    {
      "imageOffset": 813820,
      "symbol": "-[UIViewAnimationState animationDidStop:finished:]",
      "symbolLocation": 448,
      "imageIndex": 8
    },
    {
      "imageOffset": 73980,
      "symbol": "CA::Layer::run_animation_callbacks(void*)",
      "symbolLocation": 228,
      "imageIndex": 14
    },
    {
      "imageOffset": 411592,
      "symbol": "_dispatch_client_callout",
      "symbolLocation": 16,
      "imageIndex": 5
    },
    {
      "imageOffset": 69076,
      "symbol": "_dispatch_main_queue_drain",
      "symbolLocation": 928,
      "imageIndex": 5
    },
    {
      "imageOffset": 68132,
      "symbol": "_dispatch_main_queue_callback_4CF$VARIANT$mp",
      "symbolLocation": 36,
      "imageIndex": 5
    },
    {
      "imageOffset": 604184,
      "symbol": "__CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__",
      "symbolLocation": 12,
      "imageIndex": 6
    },
    {
      "imageOffset": 488600,
      "symbol": "__CFRunLoopRun",
      "symbolLocation": 2096,
      "imageIndex": 6
    }
```

</details>

**Reply by keiwando** [reply]
Thank you! Did you trigger this one directly via `Actions > Help > Report an Issue`, or via `Actions > Help > Leave a Review > Report an Issue`?
There is no multithreading involved in any part of this UI for the help options and the call stack also doesn't show anything that's not running on the main thread, so I don't think that's the reason.

**Reply by pravdomil** [reply]
there is no `Report an Issue` in Help?

<img src="https://user-images.githubusercontent.com/2387356/227804929-b19619f7-1db1-428a-84ac-493667dd31f0.jpeg" width="256">

**Reply by pravdomil** [reply]
From console:

`Unable to initialize due to + [MFMailComposeViewController canSendMail] returns NO.`

`*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'Application tried to present a nil modal view controller on target <DaVinci.AnimationEditorViewController: 0x10542d2c0>.'`

**Reply by keiwando** [reply]
Ah, yes, the fact that you don't see `Report an Issue` directly in the Help options already shows what the problem is (and then that console message you posted also confirms it). I am not showing that "Report an Issue" option if sending an e-mail is unsupported but I forgot to check it for the same option under "Leave a Review". Thanks for helping me debug this!

**Reply by pravdomil** [reply]
love you, it works!

**Comment by Byrdlnd** [comment]
It hasn’t crashed it just loads maybe one picture Sent from my iPadOn Mar 26, 2023, at 10:09 AM, Keiwan Donyagard ***@***.***> wrote:﻿
Thanks for reporting this (and for all of your other feedback!)
I've heard of this crash from some users but so far I haven't been able to reproduce it on any of my iPads at all. I am unfortunately also not seeing any related crash log. Maybe you can help me narrow down the cause of it. Could you please check if you can locate the crash log locally on your iPad in the settings app (Analytics & Improvements > Analytics Data) and send that to ***@***.***?
Does it always crash when you tap on "Report an issue" or does it work sometimes? It's supposed to open a partially pre-filled native e-mail dialog.

**Comment by pravdomil** [comment]
fixed
